File: app/src/main/java/com/example/audiobookmark/BookmarkRepository.java
package com.example.audiobookmark;

import android.content.Context;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

public class BookmarkRepository {
    private BookmarkDao dao;
    private LiveData<List<Bookmark>> allBookmarks;

    public BookmarkRepository(Context context) {
        BookmarkRoomDatabase db = BookmarkRoomDatabase.getDatabase(context);
        dao = db.dao();
        allBookmarks = dao.getAllBookmarks();
    }

    public LiveData<List<Bookmark>> getAllBookmarks() {
        return allBookmarks;
    }

    public void insert(Bookmark bookmark) {
        new InsertTask(dao).execute(bookmark);
    }

    public void delete(Bookmark bookmark) {
        new DeleteTask(dao).execute(bookmark);
    }

    private static class InsertTask extends AsyncTask<Bookmark, Void, Void> {

        private BookmarkDao dao;

        InsertTask(BookmarkDao dao) {
            this.dao = dao;
        }

        @Override
        protected Void doInBackground(final Bookmark... params) {
            dao.insert(params[0]);
            return null;
        }
    }

    private static class DeleteTask extends AsyncTask<Bookmark, Void, Void> {

        private BookmarkDao dao;

        DeleteTask(BookmarkDao dao) {
            this.dao = dao;
        }

        @Override
        protected Void doInBackground(final Bookmark... params) {
            dao.delete(params[0]);
            return null;
        }
    }
}

File: app/src/main/java/de/augmentedmind/stopit/db/BookmarkRepository.kt
package de.augmentedmind.stopit.db

import androidx.lifecycle.LiveData
import de.augmentedmind.stopit.db.Bookmark
import de.augmentedmind.stopit.db.BookmarkDao

class BookmarkRepository(private val dao: BookmarkDao) {
    val allBookmarks: LiveData<List<Bookmark>> = dao.getAllBookmarks()

    suspend fun insert(bookmark: Bookmark) {
        dao.insert(bookmark)
    }

    suspend fun delete(bookmark: Bookmark) {
        dao.delete(bookmark)
    }

}